backend/endpoints/volunteer.py

```diff
@@ -180,22 +182,50 @@ def get_volunteers_by_filters(filters, pages=0, per_page=10000):
         offset = (int(pages) - 1) * item_per_age
         if len(filters) > 0:
             flt = {}
-            toBool = {"true": True, "false": False}
-            caseS = ["first_name", "last_name"]
-            for v, k in filters.items():
-                flt[v + "__iexact" if v in caseS else v] = toBool[k.lower()] if k.lower() in toBool else k
+            case = ["zone", "role", "status"]
+
+            for key, value in filters.items():
+                if key not in case and key != "query":
+                    return jsonify({"error": key + " key can't be found"}), 400
+                if key in case:
+                    flt[key] = value
+
+            if 'query' in filters.keys():
+                obj = volunteer_keywords_search(filters['query'].split()).filter(**flt)
+            else:
+                obj = Volunteer.objects().filter(**flt)
+
+            volunteers = [json.loads(v.to_json()) for v in obj.order_by("-created_at").skip(offset).limit(item_per_age)]
 
-            obj = Volunteer.objects(**flt)
-            volunteers = [v.clean_data() for v in obj.order_by("-created_at").skip(offset).limit(item_per_age)]
             return jsonify({"list": volunteers, "count": obj.count()})
         else:
             obj = Volunteer.objects().order_by("-created_at")
-            volunteers = [v.clean_data() for v in obj.skip(offset).limit(item_per_age)]
+            volunteers = [json.loads(v.to_json()) for v in obj.skip(offset).limit(item_per_age)]
             return jsonify({"list": volunteers, "count": obj.count()})
     except Exception as error:
         return jsonify({"error": str(error)}), 400
 
 
+def volunteer_keywords_search(search_words_list, search_result=None):
```

Comment:
Let's generalize this method and move it to another file so other models can use it.

Reply:
Method is generalized, renamed to model_keywords_search and moved to utils/search.py